iceoryx2-cal/src/monitoring/process_local.rs

```diff
@@ -0,0 +1,307 @@
+        let msg = "Failed to acquire state of monitor";
+
+        let guard = fail!(from self, when PROCESS_LOCAL_STORAGE.lock(),
+            with MonitoringStateError::InternalError,
+            "{} due to a failure while acquiring the lock.", msg);
+
+        match guard.get(&self.full_name) {
+            Some(_) => Ok(State::Alive),
+            None => Ok(State::DoesNotExist),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Builder {
+    name: FileName,
+    config: Configuration,
+}
+
+impl NamedConceptBuilder<ProcessLocalMonitoring> for Builder {
+    fn new(name: &FileName) -> Self {
+        Self {
+            name: *name,
+            config: Configuration::default(),
+        }
+    }
+
+    fn config(
+        mut self,
+        config: &<ProcessLocalMonitoring as NamedConceptMgmt>::Configuration,
+    ) -> Self {
+        self.config = *config;
+        self
+    }
+}
+
+impl MonitoringBuilder<ProcessLocalMonitoring> for Builder {
+    fn token(
+        self,
+    ) -> Result<<ProcessLocalMonitoring as Monitoring>::Token, super::MonitoringCreateTokenError>
+    {
+        let msg = "Failed to create monitoring token";
+
+        let mut guard = fail!(from self, when PROCESS_LOCAL_STORAGE.lock(),
+            with MonitoringCreateTokenError::InternalError,
+            "{} due to a failure while acquiring the lock.", msg);
+
+        let full_name = self.config.path_for(&self.name);
+        if guard.contains(&full_name) {
+            fail!(from self, with MonitoringCreateTokenError::AlreadyExists,
+                "{} since the token already exists.", msg);
+        }
+
+        guard.insert(full_name);
+
+        Ok(Token {
+            name: self.name,
+            config: self.config,
+        })
+    }
+
+    fn monitor(
+        self,
+    ) -> Result<<ProcessLocalMonitoring as Monitoring>::Monitor, super::MonitoringCreateMonitorError>
+    {
+        let full_name = self.config.path_for(&self.name);
+
+        Ok(Monitor {
+            name: self.name,
+            full_name,
+        })
+    }
+
+    fn cleaner(
+        self,
+    ) -> Result<<ProcessLocalMonitoring as Monitoring>::Cleaner, super::MonitoringCreateCleanerError>
+    {
+        let msg = "Failed to create monitoring cleaner";
+
+        let guard = fail!(from self, when PROCESS_LOCAL_STORAGE.lock(),
+            with MonitoringCreateCleanerError::InternalError,
+            "{} due to a failure while acquiring the lock.", msg);
+
+        let full_name = self.config.path_for(&self.name);
+
+        match guard.get(&full_name) {
+            Some(_) => {
+                fail!(from self, with MonitoringCreateCleanerError::InstanceStillAlive,
+                    "{} since the instance is still alive.", msg);
+            }
+            None => {
+                fail!(from self, with MonitoringCreateCleanerError::DoesNotExist,
+                    "{} since the instance does not exist.", msg);
+            }
+        }
```

Comment:
Shouldn't there be a path returning an `Ok`?

Reply:
@elBoberido Not for process_local. The cleaner requires that the process is dead before cleaning up its resources. This is never given for a process local version.